controllers/precacheFsm.go

```diff
@@ -96,13 +96,11 @@ func (r *ClusterGroupUpgradeReconciler) precachingFsm(ctx context.Context,
 
 	clusterGroupUpgrade.Status.Precaching.Clusters = clusters
 
-	clusterStates := make(map[string]string)
 	for _, cluster := range clusters {
 		var currentState string
-		if len(clusterGroupUpgrade.Status.Precaching.Status) == 0 {
+		var ok bool
+		if currentState, ok = clusterGroupUpgrade.Status.Precaching.Status[cluster]; !ok {
```

Comment:
Yes, I debated whether to keep it consistent anyways. This is not needed for backup because we don't do cluster selection again once CGU is enabled. So the backup code won't get a different cluster list.

Reply:
Okay

Reply:
The other reason why I didn't make the same change for backup is that we may turn backup from one big blocking step for the whole CGU into a per cluster operation before remediating the first policy as part of batched processing.